src/wp-includes/post.php

```diff
@@ -7311,37 +7311,39 @@ function update_post_caches( &$posts, $post_type = 'post', $update_term_cache =
 
 	update_post_cache( $posts );
 
-	$post_ids = array();
-	foreach ( $posts as $post ) {
-		$post_ids[] = $post->ID;
+	if ( $update_term_cache ) {
+		update_post_term_caches( $posts, $post_type );
 	}
 
-	if ( ! $post_type ) {
-		$post_type = 'any';
+	if ( $update_meta_cache ) {
+		$post_ids = wp_list_pluck( $posts, 'ID' );
```

Comment:
I would prefer if we kept using the good old `foreach` loop here.

Reply:
It is the performance reason for this? Or just a code style one?

Reply:
I explained in the overarching PR review comment https://github.com/WordPress/wordpress-develop/pull/5134#pullrequestreview-1617974444. A `foreach` loop is more efficient and performant than instantiating a `WP_List_Util` object with all its unnecessary flexibility.